packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -124,22 +153,175 @@ const AccountRecoveryResetPassword = ({
+    let contextError: ContextValidationErrors | null = null;
+
+    try {
+      email = verificationInfo.email || '';
+
+      forceAuth = !!verificationInfo.forceAuth;
+
+      if (!verificationInfo.isValid()) {
+        supportsRecovery = false;
+        linkStatus = 'damaged';
+      } else if (!accountRecoveryKeyInfo.isValid()) {
+        supportsRecovery = false;
+      } else if (accountRecoveryKeyInfo.lostRecoveryKey === true) {
+        supportsRecovery = false;
+      }
+    } catch (err) {
+      if (err instanceof ContextValidationErrors) {
+        contextError = err;
+        linkStatus = 'damaged';
+      }
+    }
+
+    return {
+      email,
+      linkStatus,
+      forceAuth,
+      supportsRecovery,
+      contextError,
+    };
+  }
+
+  async function onSubmit(data: FormData) {
+    const password = data.newPassword;
+
+    try {
+      await account.resetPasswordWithRecoveryKey(
+        {
+          password,
+          ...verificationInfo,
+          ...accountRecoveryKeyInfo
+        }
+      );
+
+      // FOLLOW-UP: I don't see functionality in settings
+      await account.setLastLogin(Date.now());
+
+      // FOLLOW-UP: It seems like the account class will now refresh itself, so I don't think
+      //            these are necessary anymore?
+      // storageContext.set('currentAccountUid', account.uid);
+      // await account.refresh('account');
+
+      // FOLLOW-UP: No equivalent yet in settings
+      notifier.onAccountSignIn(account);
+
+      relier.resetPasswordConfirm = true;
+      logViewEvent(viewName, 'verification.success');
+
+      // FOLLOW-UP: Broker not yet implemented
+      await broker.invokeBrokerMethod('afterCompleteResetPassword', account);
+      alertSuccess();
+      navigateAway();
+    } catch (err) {
+      if (AuthUiErrors['INVALID_TOKEN'].errno === err.errno) {
+        // TODO: Is this needed? Had to add the a method to support this.
+        // BEFORE: this.logError(err);
+        logErrorEvent({ viewName, ...err });
+        setLinkStatus('expired');
+      } else {
+        // Context validation errors indicate a bad state in either the url query or
+        // maybe storage. In these cases show an alert bar and let the error keep bubbling
+        // up.
+        if (err instanceof ContextValidationErrors) {
+          alertInvalidContext(err);
+        } else {
+          logErrorEvent(err);
+          const msg = ftlMsgResolver.getMsg(
+            'unexpected-error-encountered',
+            'Unexpected Error Encountered'
+          );
+          alertBar.error(msg);
+        }
+        throw err;
+      }
+    }
+  }
+
+  async function resendLinkHandler() {
+    logViewEvent(viewName, 'account-recovery-reset-password.resend');
+
+    // FOLLOW-UP: The previous code would retry the resend several times. Is that really necessary...
+    try {
+      await account.resetPassword(state.email);
+      const msg = ftlMsgResolver.getMsg(
+        `email-resent`,
```

Comment:
Same here:
```sh
git grep -n "email-resent" | cat

packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx:285:        `email-resent`,
```

Reply:
Added to ftl file and id slightly altered.